### PR TITLE
Include tests and manual in sdist and update dependencies, and include changelog to the release

### LINKED ISSUE #5513: No more man page for version 2.1.0?
### Problem

When I download https://pypi.io/packages/source/b/beets/beets-2.1.0.tar.gz I don see any man pages anymore which was a part of https://pypi.io/packages/source/b/beets/beets-2.0.0.tar.gz, see:

```
tmp $ tar ztvf beets-2.1.0.tar.gz | grep man                                                                                              
-rwxr-xr-x  1 root     wheel        80642 Nov 22 02:50 beets-2.1.0/beets/ui/commands.py
tmp $ tar ztvf beets-2.0.0.tar.gz | grep man 
-rwxr-xr-x  1 catap    catap        80709 May 19  2024 beets-2.0.0/beets/ui/commands.py
drwxr-xr-x  2 catap    catap            0 May 30 06:13 beets-2.0.0/man
-rwxrwxrwx  1 catap    catap        22334 May 30 06:13 beets-2.0.0/man/beet.1
-rwxrwxrwx  1 catap    catap        45498 May 30 06:13 beets-2.0.0/man/beetsconfig.5
-rw-r--r--  1 catap    catap         5120 May 19  2024 beets-2.0.0/test/test_config_command.py
-rw-r--r--  1 catap    catap         3735 May 19  2024 beets-2.0.0/test/test_ui_commands.py
tmp $ 
```

### Setup

* OS: OpenBSD-current
* Python version: 3.11.10
* beets version: 2.1.0
* Turning off plugins made problem go away (yes/no):  no

### LINKED ISSUE #5518: Release workflow: populate release notes with the changelog
See the [latest release workflow run](https://github.com/beetbox/beets/actions/runs/11965116251/job/33358610317) - for some reason the changelog notes did not end up in the release notes and I had to paste them in manually.

### PR DESCRIPTION
**Pull request by snejus**
## Description

* Fixes #5513
* Fixes #5518

This PR adds the `test` folder and manuals to the package source tarball and upgrades the dependencies to fix a few vulnerabilities.

I also added a fix for the changelog not making it to the release notes.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -108,10 +111,10 @@ Other changes:
   documentation is changed, and they only check the changed files. When
   dependencies are updated (``poetry.lock``), then the entire code base is
   checked.
-* The long-deprecated `beets.util.confit` module has been removed.  This may
+* The long-deprecated ``beets.util.confit`` module has been removed.  This may
   cause extremely outdated external plugins to fail to load.
-* :doc:`plugins/autobpm`: Add plugin dependencies to `pyproject.toml` under
-  the `autobpm` extra and update the plugin installation instructions in the
+* :doc:`plugins/autobpm`: Add plugin dependencies to ``pyproject.toml`` under
+  the ``autobpm`` extra and update the plugin installation instructions in the
```
Comment (JOJ0) [inline docs/changelog.rst:117]: all these changelog fixes LGTM

File: .github/workflows/make_release.yaml [hunk 2]
```diff
@@ -40,6 +40,8 @@ jobs:
     name: Get changelog and build the distribution package
     runs-on: ubuntu-latest
     needs: increment-version
+    outputs:
+      changelog: ${{ steps.generate_changelog.outputs.changelog }}
```
Comment (JOJ0) [inline .github/workflows/make_release.yaml:44]: this is the fix/hack from the issue you linked, so LGTM

File: .github/workflows/make_release.yaml [hunk 3]
```diff
@@ -53,12 +55,13 @@ jobs:
...
 
       - name: Install dependencies
-        run: poetry install --only=release
+        run: poetry install --with=release --extras=docs
 
       - name: Install pandoc
         run: sudo apt update && sudo apt install pandoc -y
 
       - name: Obtain the changelog
+        id: generate_changelog
```
Comment (JOJ0) [inline .github/workflows/make_release.yaml:64]: and the id required for that hack, so also LGTM

File: pyproject.toml [hunk 4]
```diff
@@ -26,6 +26,7 @@ packages = [
     { include = "beets" },
     { include = "beetsplug" },
 ]
+include = ["test", "man/**/*"] # extra files to include in the sdist
```
Comment (JOJ0) [inline pyproject.toml:29]: what does ** in a path mean again?

File: pyproject.toml [hunk 5]
```diff
@@ -149,6 +150,15 @@ build-backend = "poetry.core.masonry.api"
...
 poetry = ">=1.8"
 
+[tool.poe.tasks.build]
+help = "Build the package"
+shell = """
+make -C docs man
+rm -rf man
+mv docs/_build/man .
+poetry build
+"""
```
Comment (JOJ0) [inline pyproject.toml:160]: I'm sure you tested this and made sure man build is now included in the package so also LGTM

**Review by JOJ0 — APPROVED** [review]
I'm sure you tested all this and it works so LGTM 😁Thanks!

File: .github/workflows/make_release.yaml (hunk 2 above)
Comment (JOJ0) [reply .github/workflows/make_release.yaml:44]: well, let's not call it a hack! seems to be the right thing to do: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#using-job-outputs-in-a-matrix-job

Comment (snejus) [reply .github/workflows/make_release.yaml:44]: Indeed - you can find I provided the same URL in the commit message 😅

File: pyproject.toml (hunk 4 above)
Comment (snejus) [reply pyproject.toml:29]: I tried building using simply `man`, but poetry wouldn't include the files inside this directory. `man/**/*` did the trick though.

`man/**/*` translates to _all files inside the `man` directory, recursively_. See below

image

File: pyproject.toml (hunk 5 above)
Comment (snejus) [reply pyproject.toml:160]: These instructions are also the same as the ones that were in `setup.py` previously :)

File: .github/workflows/make_release.yaml (hunk 2 above)
Comment (JOJ0) [reply .github/workflows/make_release.yaml:44]: that'ts where I got it from 😂 just wanted to clarify

**Pull request merged**